crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -103,31 +102,27 @@ impl C8yEndPoint {
     }
 }
 
-#[automock]
-#[async_trait]
-pub trait C8yJwtTokenRetriever: Send + Sync {
-    async fn get_jwt_token(&mut self) -> Result<SmartRestJwtResponse, SMCumulocityMapperError>;
-}
-
 pub struct C8yMqttJwtTokenRetriever {
     mqtt_config: mqtt_channel::Config,
 }
 
 impl C8yMqttJwtTokenRetriever {
-    pub async fn try_new(tedge_config: &TEdgeConfig) -> Result<Self, SMCumulocityMapperError> {
-        let topic = TopicFilter::new("c8y/s/dat")?;
-        let mqtt_config = tedge_config
-            .mqtt_config()?
-            .with_clean_session(true)
+    pub fn try_new(tedge_config: &TEdgeConfig) -> Result<Self, MqttConfigBuildError> {
```

Comment:
Probably `from_tedge_config()` is better name?

Reply:
Done